src/app/qgsmeasuretool.cpp

```diff
@@ -201,26 +228,55 @@ void QgsMeasureTool::canvasReleaseEvent( QgsMapMouseEvent *e )
 {
   const QgsPointXY point = e->snapPoint();
 
-  if ( mDone ) // if we have stopped measuring any mouse click restart measuring
+  if ( mDone ) // If we have stopped measuring any mouse click restart measuring
   {
     mDialog->restart();
   }
 
-  if ( e->button() == Qt::RightButton ) // if we clicked the right button we stop measuring
+  if ( e->button() == Qt::RightButton ) // If we clicked the right button we stop measuring
   {
     mDone = true;
-    mRubberBand->removeLastPoint();
-    mDialog->removeLastPoint();
+    if ( mMeasureRadius )
+    {
+      // Add point to points rb
+      addPoint( point );
+    }
+    else
+    {
+      mRubberBand->removeLastPoint();
+      mDialog->removeLastPoint();
+    }
   }
   else if ( e->button() == Qt::LeftButton )
   {
     mDone = false;
+    if ( mMeasureRadius )
+    {
+      // Consecutive left clicks will clear the rubber bands and restart measuring
+      mPoints.clear();
+      mRubberBandPoints->removeLastPoint();
+      mRubberBand->removeLastPoint();
+      mRubberBandCircle->reset( Qgis::GeometryType::Polygon );
+    }
     addPoint( point );
   }
 
   mDialog->show();
 }
 
+QgsGeometry QgsMeasureTool::createCircleGeom()
+{
+  QgsPoint tmpCntrPt = QgsPoint( mPoints.at( 0 ) );
+  QgsPoint tmpOutrPt = mRubberBand->asGeometry().vertexAt( 1 );
+  double dist = mRubberBand->asGeometry().length();
+  double az = tmpCntrPt.azimuth( tmpOutrPt );
+  QgsCircle circ = QgsCircle( tmpCntrPt, dist, az );
+  QgsPolygon *poly = circ.toPolygon( 360 );
```

Comment:
```suggestion
  QgsPolygon *poly = circ.toPolygon( 360 );
```

Why 360? FYI, it's not degree, but the number of segment. By default, it's 36, why? More than 8 years later, I've just realized that I made a mistake... It's 8 segments (as for a buffer) per quadrant, i.e. 4 x 8 ... 36... “We leave the verification as an exercise for the reader” 😅 

However, I understand it's tempting to have more segments to get a more segmented disk that really looks like a disk rather than a regular polygon. I just want to ensure a good understanding of the function and a good compromise between the number of segments, performance, and visualization.

From what I remember, 32/36 segments is usually sufficient for < 1 km. Here, indeed, we might be dealing with larger measurements. In the absence of a formula, which we can add later, I think 64 segments should be sufficient. It's up to you to tell me based on what you can observe. We need to find the right compromise.

I'll look into finding a formula on my side, perhaps taking inspiration from the sagitta deviation to determine the optimal number of segments for a given radius.